src/AppCommon/Commands/RabbitMqCommand.cs

```diff
@@ -95,28 +95,40 @@ async Task UpdateTrackers()
 
         var failCount = 0;
 
-        await Out.CountdownTimer("Data Collection Time Left", waitUntil, cancellationToken: cancellationToken, onLoopAction: async () =>
+        await Out.CountdownTimer("Data Collection Time Left", waitUntil, cancellationToken: cancellationToken, onLoopAction: () =>
         {
             if (DateTime.UtcNow > nextPollTime)
             {
-                try
-                {
-                    await UpdateTrackers();
-                }
-                catch (Exception x)
+                // Set this immediately, as loop may swing around in only 250ms
+                nextPollTime = DateTime.UtcNow + pollingInterval;
+
+                _ = Task.Run(async () =>
                 {
-                    failCount++;
-                    if (failCount >= 15)
+                    // So control can return to the loop
+                    await Task.Yield();
```

Comment:
When you offload to the worker thread pool why do you then immediately need to yield again? That makes little sense to me.

The action delegate here is returning void is that deliberate? If you would make the action a func that returns a task you could push any concerns like dropping the task or offloading into the infrastructure that you already have in place. 

Also the previous version was async void (evil I know) but would essentially have achieved the same . The synchronous part is executed as part of the loop until it his the actual IO bound operation inside UpdateTrackers and then it is fire&forget

Reply:
good comments, this aligns with many of my thoughts as well

Reply:
Yield definitely not necessary. Whoops.

So the problem was that the async void was returning immediately and it took a 100s HTTP timeout before the nextPollTime would be updated. That's what I'm trying to fix.

So yes, the onLoopAction is synchronous, but I don't want to make that async and await it because I want the loop to keep going to keep updating the timer in real time.

The Task.Run is only going to kick off every 15 minutes, and timeouts and whatnot (reduced in this same PR) inside the action should make sure they never overlap. And the Task itself handles all its own exceptions so I don't see any issue with tossing the Task away.

Shouldn't that cover things?